### PR TITLE
Change layout of Cards and Card Lists (fixes many issues)

### LINKED ISSUE #4331: Total size of channel in the import page is shown as 0
### Observed behavior

In the channel import page, the total size of the channel is shown as 0.

### Expected behavior

The total size of the channel should be shown correctly.

### User-facing consequences

Users don't know how large this channel is.

### Errors and logs

screen shot 2018-09-26 at 12 23 47 pm

### Steps to reproduce

1. Click import
2. Select `Kolibri Studio`
3. Select a channel at the import page and import the channel metadata -- at this time, the correct total size of the channel is shown
4. Go back to the channel list page
5. Click import
6. Select `Kolibri Studio`
7. Go to the imported channel page and check the total size of the channel

### Context

Kolibri version: 0.11 a7
Operating System: OSX
Browser: Chrome

### LINKED ISSUE #4394: 'view more' is shown even when there's enough space to show everything
### Observed behavior

| collapsed | expanded |
|--|--|
| image | image | 

### Expected behavior

'view more' should only be shown when there's not enough space on the card to display the text

### User-facing consequences

unnecessary clicking, wasted whitespace

### Context

0.11

### LINKED ISSUE #4395: recommend swapping card padding with spacing
### Observed behavior

currently padding is 24px and spacing between cards is 16px, which feels unbalanced

### Expected behavior

padding in cards 16px, and spacing between cards 24px:

| Before | After |
|--|--|
| image | image |

### User-facing consequences

visual tension

### Context

0.11

----

cc @khangmach thoughts?

### LINKED ISSUE #4396: font size in card descriptions is very small
### Observed behavior

Currently descriptions are only 12px

### Expected behavior

recommend bumping up to 14px. (Note that the title is 16px)

also recommend adding 8px of spacing above the 'view more' link for breathing room

| before | after | 'view more' spacing |
|--|--|--|
| image | image |  image |

### User-facing consequences

text is hard to read

### Context

0.11

----

cc @khangmach ?

### LINKED ISSUE #4397: 'select all' checkbox is not lined up with other checkboxes
### Observed behavior

image

### Expected behavior

checkboxes would be aligned

### User-facing consequences

May not immediately associate the two UIs - they feel disconnected/separate

### Context

0.11

---

cc @khangmach thoughts?

### LINKED ISSUE #4409: Channel size & number of resources reported incorrectly after the first import is cancelled
### Observed behavior

When the user selects the channel to import, and imports just the DB (_Generating the channel listing..._), but then closes the _Select content from..._ window and selects the same channel again, its size & number of resources are reported as 0. Happens on both import from local drive and Studio.

Once resources are selected, the import proceeds without issues, and on subsequent _Import more_ tries, the size is reported correctly.

ubuntu 16 04 3 start running - oracle vm virtualbox_014

### Expected behavior
Channel size & number of resources should be reported correctly even after cancelling the previous import attempts.

### User-facing consequences
Incorrect information and uncertainty about the channel to be imported.

#### Browser console
ubuntu 16 04 3 start running - oracle vm virtualbox_016

Not seeing any errors in the `server.log`
```
INFO:cherrypy.access.140653758354152:127.0.0.1 - - [12/Oct/2018:21:13:21] "GET /static/device_management_module/device_management_module-0.11.0b1.js HTTP/1.1" 304 - "http://127.0.0.1:8080/device/" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:62.0) Gecko/20100101 Firefox/62.0"
[37mINFO     Downloading data for channel id 000409f81dbe5d1ba67101cb9fed4530[0m
INFO:kolibri.core.content.management.commands.importchannel:Downloading data for channel id 000409f81dbe5d1ba67101cb9fed4530

  0%|          | 0/1343488 [00:00<?, ?it/s]
100%|██████████| 1343488/1343488 [00:00<00:00, 1653905.15it/s][37mINFO     Older version 5 of channel 000409f81dbe5d1ba67101cb9fed4530 already exists in database; removing old entries so we can upgrade to version 6[0m
INFO:kolibri.core.content.utils.channel_import:Older version 5 of channel 000409f81dbe5d1ba67101cb9fed4530 already exists in database; removing old entries so we can upgrade to version 6
[37mINFO     Importing ContentTag data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentTag data
[37mINFO     Importing ContentNode_tags data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentNode_tags data
[37mINFO     Importing ContentNode_related data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentNode_related data
[37mINFO     Importing ContentNode_has_prerequisite data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentNode_has_prerequisite data
[37mINFO     Importing ContentNode data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentNode data
[37mINFO     Importing Language data[0m
INFO:kolibri.core.content.utils.channel_import:Importing Language data
[37mINFO     Importing File data[0m
INFO:kolibri.core.content.utils.channel_import:Importing File data
[37mINFO     Importing LocalFile data[0m
INFO:kolibri.core.content.utils.channel_import:Importing LocalFile data
[37mINFO     Importing AssessmentMetaData data[0m
INFO:kolibri.core.content.utils.channel_import:Importing AssessmentMetaData data
[37mINFO     Importing ChannelMetadata data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ChannelMetadata data
[37mINFO     Setting availability of File objects based on LocalFile availability[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of File objects based on LocalFile availability
[37mINFO     Setting availability of non-topic ContentNode objects based on File availability[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of non-topic ContentNode objects based on File availability
[37mINFO     Setting availability of ContentNode objects with children for 5 levels[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for 5 levels
[37mINFO     Setting availability of ContentNode objects with children for level 5[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 5
[37mINFO     Setting availability of ContentNode objects with children for level 4[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 4
[37mINFO     Setting availability of ContentNode objects with children for level 3[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 3
[37mINFO     Setting availability of ContentNode objects with children for level 2[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 2
[37mINFO     Setting availability of ContentNode objects with children for level 1[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 1

[37mINFO     Downloading data for channel id 0e173fca6e9052f8a474a2fb84055faf[0m
INFO:kolibri.core.content.management.commands.importchannel:Downloading data for channel id 0e173fca6e9052f8a474a2fb84055faf

  0%|          | 0/1534976 [00:00<?, ?it/s]
100%|██████████| 1534976/1534976 [00:00<00:00, 1868874.27it/s][37mINFO     Importing ContentTag data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentTag data
[37mINFO     Importing ContentNode_tags data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentNode_tags data
[37mINFO     Importing ContentNode_related data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentNode_related data
[37mINFO     Importing ContentNode_has_prerequisite data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentNode_has_prerequisite data
[37mINFO     Importing ContentNode data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ContentNode data
[37mINFO     Importing Language data[0m
INFO:kolibri.core.content.utils.channel_import:Importing Language data
[37mINFO     Importing File data[0m
INFO:kolibri.core.content.utils.channel_import:Importing File data
[37mINFO     Importing LocalFile data[0m
INFO:kolibri.core.content.utils.channel_import:Importing LocalFile data
[37mINFO     Importing AssessmentMetaData data[0m
INFO:kolibri.core.content.utils.channel_import:Importing AssessmentMetaData data
[37mINFO     Importing ChannelMetadata data[0m
INFO:kolibri.core.content.utils.channel_import:Importing ChannelMetadata data
[37mINFO     Setting availability of File objects based on LocalFile availability[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of File objects based on LocalFile availability
[37mINFO     Setting availability of non-topic ContentNode objects based on File availability[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of non-topic ContentNode objects based on File availability
[37mINFO     Setting availability of ContentNode objects with children for 5 levels[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for 5 levels
[37mINFO     Setting availability of ContentNode objects with children for level 5[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 5
[37mINFO     Setting availability of ContentNode objects with children for level 4[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 4
[37mINFO     Setting availability of ContentNode objects with children for level 3[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 3
[37mINFO     Setting availability of ContentNode objects with children for level 2[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 2
[37mINFO     Setting availability of ContentNode objects with children for level 1[0m
INFO:kolibri.core.content.utils.annotation:Setting availability of ContentNode objects with children for level 1
```

### Steps to reproduce
1. Select a new channel to import
2. When the channel content tree loads, close the  _Select content from..._ window without actually importing anything
3. Select the same channel again
4. Observe the total size and number of resources values

### Context

 * Kolibri version: `0.11.0b1`
 * Operating system: Ubuntu 16.04
 * Browser: both FF & Chromium

cc @indirectlylit @rtibbles

### LINKED ISSUE #4435: Channel availability cached on channel import
### Observed behavior
When a channel has previously been imported, then is deleted, and then the process for importing for that channel is begun again, the previous import status of the content within the granular selection interface is displayed.

### Expected behavior
Granular import status should reflect the actual state.

### User-facing consequences
Needs a refresh to force the view.

### Steps to reproduce
Delete a channel that has been imported.
Without refreshing then try to reimport that channel.

### Context
* v0.11.0-beta1-dev

### PR DESCRIPTION
**Pull request by jonboiser**
### Summary

1. Increase max-height for descriptions. Fixes #4394.
1. Force fetch all ContentNodeGranular requests in import/export. Fixes #4435
1. Increase font size in Lesson cards (per #4403). Also reduces inner-card padding to 16px. Fixes #4396.
1. Move cards with checkboxes to center. Lists with no check-able cards are also aligned to the left. Fixes #4397 and #4391
1. Increase bottom margin on cards to 24px. Fixes #4395.
1. Makes sure to append total resources + file size when normalizing ChannelMetadata objects during import/export. Fixes #4409 and Fixes #4331

*Card Layouts*

All with checkboxes, none with checkboxes, some with checkboxes. Screenshot taken before increase inter-card spacing to 24px.

screen shot 2018-10-15 at 3 30 30 pm
screen shot 2018-10-15 at 3 30 16 pm
screen shot 2018-10-15 at 3 30 10 pm

### Reviewer guidance

1. Check cards' appearance in Lesson + Exam workflows with various channels
1. Check to see that granular import/export workflow accurately reflects the state of the server.
1. Check to see that #4409 and #4331 are addressed when importing from kolibri studio or peer server.
…

### References

…

----

### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

**Comment by indirectlylit** [comment]
FYI, this might interact with #4396. @JasonTame was working on that in #4403

File: kolibri/plugins/coach/assets/src/views/lessons/LessonResourceSelectionPage/ContentCardList.vue [hunk 1]
```diff
@@ -15,7 +15,7 @@
         :key="content.id"
       >
         <KCheckbox
-          v-if="!contentHasCheckbox(content)"
+          v-if="contentHasCheckbox(content)"
```
Comment (christianmemije) [inline kolibri/plugins/coach/assets/src/views/lessons/LessonResourceSelectionPage/ContentCardList.vue:18]: :+1:

File: kolibri/core/assets/src/views/TextTruncator.vue [hunk 2]
```diff
@@ -105,6 +110,7 @@
 <style lang="scss" scoped>
 
   .ar {
+    margin-top: 8px;
```
Comment (christianmemije) [inline kolibri/core/assets/src/views/TextTruncator.vue:113]: Why this top-margin?

Comment (jonboiser) [reply kolibri/core/assets/src/views/TextTruncator.vue:113]: This was from Jason Tame's PR to add more spacing between the View More/Less button and the description.

Comment (indirectlylit) [reply kolibri/core/assets/src/views/TextTruncator.vue:113]: since it's no longer just 'align right' maybe give it a more descriptive class name?

**Review by indirectlylit — APPROVED** [review]
this looks awesome! great job

**Comment by indirectlylit** [comment]
I am still reproducing #4409, so we'll leave that issue open:

import

**Pull request merged**